checker.go

```diff
@@ -73,6 +73,34 @@ func (c *equalsChecker) Check(got interface{}, args []interface{}, note func(key
 	return nil
 }
 
+// NilError is a Checker that checks whether an error value is nil.
```

Comment:
I'd call it `IsNilError`:
- it is similar to IsNil that we already have;
- it will show up in autocompletion as an alternative to IsNil;
- all other checkers are 3rd person verbs.